include/networkit/components/StronglyConnectedComponents.hpp

```diff
@@ -1,70 +1,143 @@
+
 public:
-    StronglyConnectedComponents(const Graph& G, bool iterativeAlgo=true);
+    /**
+     * Determines the strongly connected components of a directed graph using Tarjan's algorithm.
+     *
+     * @param G Graph A directed graph.
+     */
+    StronglyConnectedComponents(const Graph &G);
 
     /**
-     * This method determines the connected components for the graph g
-     * (by default: iteratively).
+     * Determines the strongly connected components of a directed graph using Tarjan's algorithm.
+     *
+     * @param G Graph A directed graph.
+     * @param iterativeAlgo bool Whether to use the iterative algorithm or the recursive one.
+     *
+     * This constructor has been deprecated because the recursive algorithm has been deprecated.
+     */
+    TLX_DEPRECATED(StronglyConnectedComponents(const Graph &G, bool iterativeAlgo));
+
+    /**
+     * Runs the algorithm.
      */
-    void run();
+    void run() override;
 
     /**
      * This method determines the connected components for the graph g
      * (iterative implementation).
+     *
+     * This method is deprecated, run() already uses the iterative implementation.
      */
-    void runIteratively();
+    void TLX_DEPRECATED(runIteratively());
 
     /**
      * This method determines the connected components for the graph g
      * (recursive implementation).
+     *
+     * This method is deprecated, use run().
+     */
+    void TLX_DEPRECATED(runRecursively());
+
+    /**
+     * Return the number of connected components.
+     *
+     * @return count The number of components of the graph.
+     */
+    count numberOfComponents() const {
+        assureFinished();
+        return componentIndex;
+    }
+
+    /**
+     * Returns the component of the input node.
+     *
+     * @param[in] u The input node.
+     * @return count The index of the component of the input node.
      */
-    void runRecursively();
+    index componentOfNode(node u) const {
+        assureFinished();
+        assert(G->hasNode(u));
+        return component[u];
+    }
 
     /**
-     * This method returns the number of connected components.
+     * Return a Partition object that represents the components.
+     *
+     * @return Partition Partitioning of the strongly connected components.
      */
-    count numberOfComponents();
+    Partition getPartition() const {
+        assureFinished();
+        return Partition(component);
+    }
 
     /**
-     * This method returns the the component in which node query is situated.
+     * Return a map with the component indexes as keys, and their size as values.
      *
-     * @param[in]	query	the node whose component is asked for
+     * @return std::map<index, count> Map with components indexes as keys, and their size as values.
      */
-    count componentOfNode(node u);
+    std::map<node, count> getComponentSizes() const {
+        assureFinished();
+        std::map<node, count> map;
```

Comment:
As component ids are continues, wouldn't it be a lot more efficient to use `std::vector<count>` here? Or is this just to be compatible with the `Partition` interface? But even then, it should be more efficient to first aggregate the values in a `std::vector<count>` and only then convert this to a `std::map<node, count>`.

Reply:
I am using a `std::map<node, count>` to be compatible with both `Partition` and `ConnectedComponents`, we might change those APIs or provide new ones in a separate PR. Apart from that it's true that using a `std::vector<count>` to aggregate values would be better, I implemented this change.